### ISSUE TITLE
Add way to enable OVERLAP_SIMPLE or OVERLAP_COMPOUND

### ISSUE THREAD

**Issue by moyogo** [googlefonts/fontmake#681]
There doesn’t seem to be a way to enable OVERLAP_SIMPLE in `ttglyph.flags` or OVERLAP_COMPOUND in `ttglyph.components[0].flags`. These are required for correct handling of overlaps on macOS.

Could `--keep-overlaps` enable these when generating `ttf`?

See https://github.com/TypeNetwork/Vollkorn-Typeface/issues/8#issuecomment-655679836

**Comment by moyogo** [comment]
See also https://github.com/fonttools/fonttools/pull/1518

**Comment by anthrotype** [comment]
yeah, makes sense that we enable those flags when --keep-overlaps is set. Environments that don't use the flag will simply ignore it. Can you send a PR?

**Comment by moyogo** [comment]
Actually, for Vollkorn instances, the overlaps are removed for simple glyphs but the overlap issue occurs with composite glyphs. So the `OVERLAP_COMPOUND` bit should be enabled some other way.

**Comment by madig** [comment]
Hum. I suppose one could check if the bounding boxes of contours or components overlap and set `OVERLAP_SIMPLE` or `OVERLAP_COMPOUND` respectively?

**Comment by m4rc1e** [comment]
Thought I'd just post a similar issue to this thread.

We're getting requests from the Android team to enable bit 6 for VF glyphs which contain overlapping contours, https://github.com/google/fonts/issues/4405.

Here's a previous discussion related to this issue and the implications for setting it may have,
https://github.com/fonttools/fonttools/issues/2059

Since I need to enable this bit for our VF font collection, I plan to write a very primitive solution since my graphics/2d knowledge is weak. I don't want to enable this bit for every glyph since it will hammer FreeType. My plan is to detect which glyphs have overlaps by duplicating a glyph's outlines, removing the overlaps on the duplicates and then diffing the areas. This is probably too computationally expensive to include in fontmake so I'll just write it as a gftools script to begin with.

**Comment by madig** [comment]
@m4rc1e nope, use the code in https://github.com/google/fonts/issues/4405#issuecomment-1077650700 and integrate it into ufo2ft if it's not there already, we use it in production and it completes in no time. I think I barely measured any compilation time increase.

**Comment by m4rc1e** [comment]
Thanks! Have you measured what impact this will have on FreeType?

**Comment by madig** [comment]
Nope! But you need it for overlaps anyway so you have to pay the price 😬

**Comment by madig** [comment]
> Do we have data on the range of % total glyphs with overlapping contours in "typical" VF families? 

I got two figures here: https://github.com/google/fonts/issues/4405#issuecomment-1079185880

**Comment by chrissimpkins** [comment]
> > Do we have data on the range of % total glyphs with overlapping contours in "typical" VF families?
> 
> I got two figures here: [google/fonts#4405 (comment)](https://github.com/google/fonts/issues/4405#issuecomment-1079185880)

The answer is few, many, and most.  :)

**Comment by tphinney** [comment]
In Science Gothic and an unnamed Google internal project I am working on, the answer is on the order of 1% or 2%.

Mind you, it is currently going up, and will be higher in future projects, because I know that a solution to make the rendering perfect is in the works.

I have avoided it in the past, only choosing overlap as a sort of lesser evil. But if it can render correctly, then I will use it more.
